lib/PlacesProvider.js

```diff
@@ -115,7 +118,38 @@ Links.prototype = {
     },
 
     QueryInterface: XPCOMUtils.generateQI([Ci.nsINavHistoryObserver,
-                        Ci.nsISupportsWeakReference])
+                                           Ci.nsISupportsWeakReference])
+  },
+
+  /**
+   * A set of functions called by @mozilla.org/browser/nav-bookmarks-service
+   * All bookmark events are emitted from this object.
+   */
+  bookmarksObserver: {
+    onItemAdded(id, folderId, index, type) {
+      if (type == Bookmarks.TYPE_BOOKMARK) {
+        gLinks.getBookmark({id}).then(bookmark => {
+          gLinks.emit("bookmarkAdded", bookmark);
+        });
+      }
+    },
+
+    onItemRemoved(id, folderId, index, type, uri) {
+      if (type == Bookmarks.TYPE_BOOKMARK) {
+        gLinks.emit("bookmarkRemoved", {bookmarkId: id, url: uri.spec});
+      }
+    },
+
+    onItemChanged(id, property, isAnnotation, value, lastModified, type) {
+      if (type == Bookmarks.TYPE_BOOKMARK) {
```

Comment:
Ah! We need to make sure we catch that. We're using the Mozilla style guide, but it's very lax, to make sure gecko at least has a chance of passing linting